assets/app/scripts/services/navigate.js

```diff
@@ -46,6 +46,67 @@ angular.module("openshiftConsole")
        */
       toNextSteps: function(name, projectName){
         $location.path("project/" + encodeURIComponent(projectName) + "/create/next").search("name", name);
+      },
+
+      // Resource is either a resource object, or a name.  If resource is a name, kind and namespace must be specified
+      // Note that builds and deployments can only have their URL built correctly (including their config in the URL)
+      // if resource is an object, otherwise they will fall back to the non-nested URL.
+      // TODO - if we do ever need to create a build URL without the build object but with a known build (deployment) 
+      // name and buildConfig (deploymentConfig) name, then we will need either a specialized method for that, or an
+      // additional opts param for extra opts.
+      resourceURL: function(resource, kind, namespace) {
+        if (!resource || (!resource.metadata && (!kind || !namespace))) {
+          return null;
+        }
+
+        // normalize based on the kind of args we got
+        if (!kind) {
+          kind = resource.kind;
+        }
+        if (!namespace) {
+          namespace = resource.metadata.namespace; 
+        }
+        namespace = encodeURIComponent(namespace);
+
+        var name = resource;
+        if (resource.metadata) {
+          name = resource.metadata.name;
+        }
+
+        name = encodeURIComponent(name);
```

Comment:
fyi encoding namespace and name up here so i didnt have to write encodeURIComponent ten times